File: frontend/app/types/request.ts
export type RequestStatus =
  | "pending"
  | "in_progress"
  | "approved"
  | "rejected";

export type PriorityLevel = "low" | "medium" | "high" | "urgent";

export type CreateRequestDto = {
  service_type: string;
  description: string;
  is_urgent: boolean;
  student_name?: string;
};

export type CreateRequestResponse = {
  request_id: string;
  priority: string;
  queue_position: number;
  predicted_wait_minutes: number;
};

export type ListRequestItem = {
  request_id: string;
  student_name: string;
  service_type: string;
  priority_level: PriorityLevel;
  status?: string | null;
  created_at?: string;
  queue_position?: number;
};

File: backend/app/schemas/request.py
from typing import Optional

from pydantic import BaseModel, Field


class RequestCreate(BaseModel):
    service_type: str = Field(..., min_length=1)
    description: str = Field(..., min_length=10)
    is_urgent: bool = False
    student_name: Optional[str] = ""


class RequestUpdate(BaseModel):
    status: Optional[str] = None
    description: Optional[str] = None
    priority_level: Optional[str] = None


class RequestResponse(BaseModel):
    request_id: str
    student_name: str
    service_type: str
    description: str

    priority_level: str
    priority_score: float

    predicted_wait_minutes: float
    queue_position: Optional[int] = None
    status: Optional[str] = None

